app/validators/questionnaire_schema.py

```diff
@@ -143,6 +143,19 @@ def __init__(self, schema):
             '$..blocks[?(@.type=="ListCollector")].for_list', self.schema
         )
         self.list_names = self.list_collector_names + self.supplementary_lists
+        self.list_names_by_repeating_block_id = {
```

Comment:
Maybe it's just me but two for loops in this comprehension, I would do just two normal loops for readability's sake. Same  the next one below, but up to you and other devs.

Reply:
Yeah I think the second of these two properties is probably a bit much, I've changed it to a cached property with a normal for loop. For the first one, I've made list collectors a shared variable so now its still two loops but I think easier to follow, let me know what you think? can always do another cached property if preferable

Reply:
Thanks, much easier to follow now.